controllers/actions.github.com/autoscalingrunnerset_controller.go

```diff
@@ -154,7 +160,7 @@ func (r *AutoscalingRunnerSetReconciler) Reconcile(ctx context.Context, req ctrl
 
 	// Make sure the runner group of the scale set is up to date
 	currentRunnerGroupName, ok := autoscalingRunnerSet.Annotations[runnerScaleSetRunnerGroupNameKey]
-	if !ok || !strings.EqualFold(currentRunnerGroupName, autoscalingRunnerSet.Spec.RunnerGroup) {
+	if !ok || (len(autoscalingRunnerSet.Spec.RunnerGroup) > 0 && !strings.EqualFold(currentRunnerGroupName, autoscalingRunnerSet.Spec.RunnerGroup)) {
```

Comment:
`autoscalingRunnerSet.Spec.RunnerGroup` can be empty, which means we are using the `default` runner group.

Reply:
Do we need to check if annotation exists? The only way we can get to this part is that we created a scale-set and in that method, we are adding the annotation. So the only way to hit this part is by having scale set created, and with that, the annotation: https://github.com/actions/actions-runner-controller/blob/d0c6a96fca1589233c7de751fb4bc5703e430475/controllers/actions.github.com/autoscalingrunnerset_controller.go#L359

Reply:
I will add a check since technically user can remove it by accident.

Reply:
Actually, should the `!ok||` take care the case?

Reply:
Yeah, I think we're good as is.